rr_viz/src/rr_viz_tabs.py

```diff
@@ -80,46 +84,86 @@ def __init__(self, parent):
         self.main_window_layout.addWidget(self.main_window_tabs)
         self.setLayout(self.main_window_layout)
 
-        # Set up topic subscribers
-        self.battery_level_sub_name = "/vesc_driver/battery"
-        self.battery_level_sub = rospy.Subscriber(self.battery_level_sub_name, BatteryState, self.battery_level_update)
-
         # Get Battery levels 
         self.good_voltage = rospy.get_param("/battery/good_voltage", "31")
         self.ok_voltage = rospy.get_param("/battery/ok_voltage", "26")
         self.low_voltage = rospy.get_param("/battery/low_voltage", "22")
+        self.battery_state = False
 
-        # Variables required to detect if connected to ROS MASTER
-        self.ros_loss_triggered = True
-        self.rosMasterIP = '192.168.10.100'
-        self.subprocess_command = 'fping -nV -t 50 ' + self.rosMasterIP
-        self.fp_status = "alive"  
+        # Set up topic subscribers
+        self.battery_level_sub_name = rospy.get_param("/battery/level", "/vesc_driver/battery")
+        self.battery_level_sub = rospy.Subscriber(self.battery_level_sub_name, BatteryState, self.battery_level_update)
 
+        # Set up core connection variables
+        self.core_clock_sub_name = rospy.get_param("/core_clock_sub","/core_clock_topic")
+        self.core_clock_sub = rospy.Subscriber(self.core_clock_sub_name, Time, self.core_clock_cb)
+        self.last_received_time = Time()
+
+        self.read_time = Time()
+        self.no_time_change = 0
+        self.latch = 0
+
+        # Core timer
+        core_clock_pub_freq = rospy.get_param("~publish_frequency", 10)
+        freq = float(core_clock_pub_freq) - (float(core_clock_pub_freq) / 100 * 10)
```

Comment:
I don't understand why we substract 10% of the frequency? I think this bit could use a comment if it is to stay

Reply:
If I'm reading this correctly it will cause the core_clock_timer to be called quicker than the publish frequency. Is this the desired functionality, or would we want the timer to slightly lag behind the core publishes?

Reply:
@msadowski From your first comment on this PR you said to implement another timeout called every clock period + 10%. So when i was doing this maths if i added the 10% at this point it is still frequency and when it gets divided by 1 to convert to period the additional timeout was being called first, hence the subtraction so the additional timer timeout 10% after the clock cb would. I will add a comment in the code saying this